lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -0,0 +1,517 @@
+        Returns:
+            the path of the file from the workload's perspective
+        """
+        return f"{self.tls_directory_path}/{file.name.lower()}.pem"
+
+    def get_tls_file_path_in_charm(self, file: File) -> str:
+        """Return the requested file's location in the charm (not in the workload).
+
+        This path would typically be: /var/lib/juju/storage/certs/0/{file}.pem
+
+        Args:
+            file: a File object that determines which file path to return
+        Returns:
+            str: path
+        Raises:
+            VaultCertsError: If the CA certificate is not found
+        """
+        storage = self.charm.model.storages
+        if "certs" not in storage:
+            raise VaultCertsError()
+        if not storage["certs"]:
+            raise VaultCertsError()
+        cert_storage = storage["certs"][0]
+        try:
+            storage_location = cert_storage.location
+        except ModelError as e:
+            # Seems to happen when the storage is still being set up
+            logging.warning("Could not get storage location: %s", e)
+            raise VaultCertsError()
+        return f"{storage_location}/{file.name.lower()}.pem"
+
+    def tls_file_available_in_charm(self, file: File) -> bool:
+        """Return whether the given file is available in the charm.
+
+        Args:
+            file: a File object that determines which file to check
+        Returns:
+            bool: True if file exists
+        """
+        try:
+            file_path = self.get_tls_file_path_in_charm(file)
+            return os.path.exists(file_path)
+        except VaultCertsError:
+            return False
+
+    def _get_ca_certificate_secret(self) -> Tuple[str, str]:
+        """Get the vault CA certificate secret.
+
+        Returns:
+            The CA private key and certificate as a tuple
+        """
+        juju_secret = self.charm.model.get_secret(label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+        content = juju_secret.get_content(refresh=True)
+        return content["privatekey"], content["certificate"]
+
+    def _set_ca_certificate_secret(
+        self,
+        private_key: str,
+        certificate: str,
+    ) -> None:
+        juju_secret_content = {
+            "privatekey": private_key,
+            "certificate": certificate,
+        }
+        if not self.ca_certificate_secret_exists():
+            self.charm.app.add_secret(juju_secret_content, label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+            logger.debug("Vault CA certificate secret set")
+            return
+        secret = self.charm.model.get_secret(label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+        secret.set_content(juju_secret_content)
+
+    def ca_certificate_secret_exists(self) -> bool:
+        """Return whether CA certificate is stored in secret."""
+        try:
+            ca_private_key, ca_certificate = self._get_ca_certificate_secret()
+            if ca_private_key and ca_certificate:
```

Comment:
You can return this value directly and remove line 391

Reply:
done in #499